autofaiss/indices/distributed.py

```diff
@@ -261,8 +264,13 @@ def run(
     # broadcast the index bytes
     trained_index_bytes = get_bytes_from_index(faiss_index)
     broadcast_trained_index_bytes = ss.sparkContext.broadcast(trained_index_bytes)
-    nb_vectors = embedding_reader.count
-    nb_batches = math.ceil(nb_vectors / batch_size)  # use math.ceil to make sure that we cover every vector
+    sc = ss._jsc.sc()  # pylint: disable=protected-access
+    n_workers = len([executor.host() for executor in sc.statusTracker().getExecutorInfos()]) - 1
```

Comment:
Is `len(sc.statusTracker().getExecutorInfos()) - 1` sufficiant  here ?

Reply:
good point, fixed